### PR TITLE
Make certificate path length configurable

### PR DESCRIPTION
**Pull request by Ioana37**

### CODE REVIEW

File: KeyVaultCa.Core/EstConfiguration.cs [hunk 1]
```diff
@@ -7,5 +7,7 @@ public class EstConfiguration
         public string IssuingCA { get; set; }
 
         public int CertValidityInDays { get; set; } = 365;
```
Comment (MagdaPaj) [inline KeyVaultCa.Core/EstConfiguration.cs:9]: is that still needed? Won't this be taken from appsettings.json?

File: README.md [hunk 2]
```diff
@@ -36,7 +36,9 @@ The following common block must be filled in, for all usages of the projects.
 ```
 "KeyVault": {
     "KeyVaultUrl": "<Key Vault URL>",
-    "IssuingCA": "<Name of the issuing certificate in KeyVault.>"
+    "IssuingCA": "<Name of the issuing certificate in KeyVault.>",
+    "CertValidityInDays": "<Validity period for issued certificates (maximum is 365 days)>",
```
Comment (MagdaPaj) [inline README.md:40]: who or where this 365 maximum is enforced?

File: README.md [hunk 3]
```diff
@@ -36,7 +36,9 @@ The following common block must be filled in, for all usages of the projects.
...
+    "CertPathLength": "<Path length of the certificate chain which gives the maximum number of non-self-issued intermediate certificates that may follow this certificate in a valid certification path. For the Root CA stored in KV, the default is 1, for the others is 0.>"
```
Comment (MagdaPaj) [inline README.md:41]: So the current default value in the appsettings is 1, so it's correct for the Root CA stored in KV. Will user know that he/she should change the value to 0 when using the others?

File: KeyVaultCa.Core/EstConfiguration.cs (hunk 1 above)
Comment (MagdaPaj) [reply KeyVaultCa.Core/EstConfiguration.cs:9]: sorry, I wasn't specific enough. I was referring to the initialization

Comment (Ioana37) [reply KeyVaultCa.Core/EstConfiguration.cs:9]: Initially, I wanted to consider this property as optional and had a default value assigned, but with the current way of configuring, it should not be needed anymore. This value would anyway be overridden by what is specified in other settings.

**Review by MagdaPaj — APPROVED** [review]

**Pull request merged**